craft_providers/actions/snap_installer.py

```diff
@@ -180,6 +180,94 @@ def _get_host_snap(snap_name: str) -> Iterator[pathlib.Path]:
         yield snap_path
 
 
+def _get_assertion(query: List[str]) -> bytes:
+    """Get an assertion from snapd.
+
+    :param query: assertion query to pass to `snap known`
+    :returns: assertion data
+    :raises SnapInstallationError: if 'snap known' call fails
+    """
+    command = snap_cmd.formulate_known_command(query=query)
+    logger.debug("Executing command on host: %s", shlex.join(command))
+    try:
+        return subprocess.run(command, capture_output=True, check=True).stdout
+    except subprocess.CalledProcessError as error:
+        raise SnapInstallationError(
+            brief="failed to get assertions for snap",
+            details=details_from_called_process_error(error),
+        ) from error
+
+
+@contextlib.contextmanager
+def _get_assertions_file(
+    snap_name: str, snap_id: str, snap_revision: str
+) -> Iterator[pathlib.Path]:
+    """Get an assertion file for a snap.
+
+    :param snap_name: Name of snap to inject
+    :param snap_id: ID of the snap
+    :param snap_revision: Revision of the snap
+
+    :yields: path to temporary assertion file
```

Comment:
this docstring is confusing here, as this is not a "generator that yields values", but a "context manager that will set the temporary assertion file as the target"... right?